awx_collection/test/awx/test_completeness.py

```diff
@@ -0,0 +1,251 @@
+    'tower_workflow_template',
+]
+
+# Global module parameters we can ignore
+ignore_parameters = [
+    'state', 'new_name',
+    # The way we implemented notifications was to add these fileds to modules which can notify
+    'notification_templates_approvals',
+    'notification_templates_error',
+    'notification_templates_started',
+    'notification_templates_success',
+]
+
+# Some modules take additional parameters that do not appear in the API
+# Add the module name as the key with the value being the list of params to ignore
+no_api_parameter_ok = {
+    # The credential module used to take a lot of parameters which are now combined into "inputs"
+    'tower_credential': [
+        'authorize', 'authorize_password', 'become_method', 'become_password', 'become_username', 'client',
+        'domain', 'host', 'kind', 'password', 'project', 'secret', 'security_token', 'ssh_key_data',
+        'ssh_key_unlock', 'subscription', 'tenant', 'username', 'vault_id', 'vault_password',
+    ],
+    # The wait is for wheter or not to wait for a projec update on change
+    'tower_project': ['wait'],
+    # Existing_token and id are for working with an exisitng tokens
+    'tower_token': ['existing_token', 'existing_token_id'],
+    # Children and hosts are how tower_group deals with associations
+    'tower_group': ['children', 'hosts'],
+    # Credential and Vault Credential are a legacy fields for old Towers
+    # Credentials/labels/survey spec is now how we handle associations
+    # We take an organization here to help with the lookups only
+    'tower_job_template': ['credential', 'vault_credential', 'credentials', 'labels', 'survey_spec', 'organization'],
+    # Always_nodes, failure_nodes, success_nodes, credentials is how we handle associations
+    # Organization is how we looking job templates
+    'tower_workflow_job_template_node': ['always_nodes', 'failure_nodes', 'success_nodes', 'credentials', 'organization'],
+    # Survey is how we handle associations
+    'tower_workflow_job_template': ['survey'],
+    # These fields get wrapped into notification_configuration
+    'tower_notification_template': [
+        'account_sid', 'account_token', 'channels', 'client_name', 'color', 'from_number', 'headers', 'host',
+        'message_from', 'nickname', 'notify', 'password', 'port', 'recipients', 'sender', 'server',
+        'service_key', 'subdomain', 'targets', 'to_numbers', 'token', 'url', 'use_ssl', 'use_tls', 'username',
+    ]
+}
+
+# When this tool was created we were not feature complete. Adding something in here indicates a module
+# that needs to be developed. If the module is found on the file system it will auto-detect that the
+# work is being done and will bypass this check. At some point this module should be removed from this list.
+needs_development = [
+    'tower_ad_hoc_command', 'tower_application', 'tower_instance_group', 'tower_inventory_script',
```

Comment:
`tower_inventory_script` is for a deprecated endpoint, so you could move this over to the ignore list instead.